src/hooks/useRankingParams.ts

```diff
@@ -0,0 +1,37 @@
+import { useSearchParams } from 'react-router-dom';
+
+const PARAM_KEYS = {
+  TARGET_TYPE: 'targetType',
+  RANK_TYPE: 'rankType',
+} as const;
+
+const DEFAULT_VALUES = {
+  targetType: 'ALL',
+  rankType: 'MANY_WISH',
+} as const;
+
+export const useRankingParams = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const targetType =
+    searchParams.get(PARAM_KEYS.TARGET_TYPE) || DEFAULT_VALUES.targetType;
+  const rankType =
+    searchParams.get(PARAM_KEYS.RANK_TYPE) || DEFAULT_VALUES.rankType;
+
+  const setTargetType = (value: string) => {
```

Comment:
```suggestion
  const setTargetType = (value: keyof typeof PARAM_KEYS) => {
```

로 하면 타입이 더 명확하게 추론돼서 사용하는 동료들이 편리하게 사용할 수 있을 거에요!

[예시]
<img width="554" height="115" alt="image" src="https://github.com/user-attachments/assets/5de7ff46-9fda-4ea4-857e-18330e4ff513" />

그리고 hook으로 분리하니까 훨씬 깔끔하군요!

Reply:
피드백을 수용하려 관련 파일들을 살펴보던 중 TargetType, RankType을 설정하는 것에 있어서 하드코딩이나 비슷한 코드 발견 등 중복되는 부분들이 있는 것을 인지했습니다! 이에 나중에 관리하기 쉬우려면 한 번 중복을 모아주는 시간이 필요하다고 생각했고 "타입 명확화 & 중복 코드 제거"를 목표로 삼아 피드백을 반영해보았습니다!

  - [x] rankingParams.ts를 만들어서 Ranking 부분과 관련있는 하드코딩된 부분과 타입을 상수처리해주었습니다.
  - [x] RankingSort, RankingFilter에서 rankingParams에 정의된 상수를 가져와 사용하도록 해주었습니다.
  - [x] useRankingParams.ts에서도 타입 정의 과정을 수행했습니다.

위의 과정을 통해 rank와 관련된 변수들을 관리하기 쉽고 타입을 명확하게 추론하도록 해주었습니다!

Reply:
척하면 척이군요.. 고생하셨습니다. 

앞으로 작업이 훨씬 수월해지실 거에요!